components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/openjdk/nashorn/JsOpenJdkNashornStep.java

```diff
@@ -125,4 +146,21 @@ private List<Map<String, String>> getOptions() {
                 })));
         return optionsList;
     }
+
+    protected String getAuthenticatedAuthenticatorOfCurrentStep(AuthenticationContext context) {
+
+        if (context.getSequenceConfig() == null) {
+            //Sequence config is not yet initialized
+            return null;
+        }
+
+        StepConfig stepConfig = context.getSequenceConfig().getStepMap()
+                .get(context.getCurrentStep());
+        if (stepConfig != null) {
+            return stepConfig.getAuthenticatedAutenticator().getName();
+        }
+        return null;
```

Comment:
Shall we simplify this

Reply:
already done by https://github.com/wso2/carbon-identity-framework/pull/4662/commits/c0d6dbd72bfc8d0e4ec327f365e304b2cb9051de